Cargo shuttle now doesn't sell people

LuisDSG opened the pull request:
'Edited the CargoShuttle script and separated its item scanner from the "UnloadCargo" function before attaching it to a public "SearchForObjectsOnShuttle" function that returns a Transform Object
-Added a "LifeformChecker" function in CargoManager that looks for any players or animals on the shuttle.
-Modified the "CallShuttle" function on CargoManager so that, using "LifeformChecker", it avoids flying off from the station with a player or animal.

### Purpose
Fixes the issue that allowed the cargo shuttle to fly off from the station with characters on board and erase them afterwards by preventing the shuttle from flying as long as any object that could be alive is aboard.
### Notes:
None

LuisDSG committed:
Shuttle now doesn't sell people

Edited the CargoShuttle script and separated its item scanner from the "UnloadCargo" function before attaching it to a public "SearchForObjectsOnShuttle" function that returns a Transform Object
-Added a "LifeformChecker" function in CargoManager that looks for any players or animals on the shuttle.
-Modified the "CallShuttle" function on CargoManager so that, using "LifeformChecker", it avoids flying off from the station with a player or animal.

ReiderJack commented on UnityProject/Assets/Scripts/Managers/CargoManager.cs:
```diff
@@ -59,6 +59,23 @@ private void OnDisable()
+	/// Checks for any Lifeforms (dead or alive) that might be aboard the shuttle
+	/// </summary>
+	/// <returns>true if a lifeform is found, false if none is found</returns>
+	bool LifeformChecker()
```
Good rule is to call methods with a verb. In this case for example: `CheckLifeforms`

ReiderJack replied:
Also we specify access modifiers. At least doing it lately. So please add `private`

ReiderJack commented on UnityProject/Assets/Scripts/Managers/CargoManager.cs:
```diff
@@ -59,6 +59,23 @@ private void OnDisable()
+		Transform ObjectHolder = CargoShuttle.Instance.SearchForObjectsOnShuttle();
+		for (int i = 0; i < ObjectHolder.childCount; i++)
+		{
+			if (ObjectHolder.GetChild(i).GetComponent<LivingHealthBehaviour>() != null)
```
This is really expensive way to check if an object is a player. Did you try to check by layers?
<https://docs.unity3d.com/Manual/Layers.html> 
There is a Player layer.

ReiderJack replied:
`if(((1<<gameObject.layer) & layers) == 0)` return `true`;
Where layers has Player layer. 
<https://docs.unity3d.com/ScriptReference/LayerMask.html>

ReiderJack replied:
You can do ` LayerMask.GetMask("Player")`, but cache it before iterating in `for` to not call it everytime.

ReiderJack replied:
Also you can iterate without using `for`
```
foreach (Transform child in transform)
{
  //child is your child transform
}
```

LuisDSG committed:
Shuttle Lifeform checking edits

-"LifeformChecker" renamedto "CheckLifeforms"
-CheckLifeforms now searches for lifeforms on the shuttle by looking up the layer values of the items inside instead of using GetComponent.

LuisDSG committed:
Update CargoManager.cs

-Changed the "for" loop in "CheckLifeforms" to a "foreach" loop

ReiderJack commented on UnityProject/Assets/Scripts/Managers/CargoManager.cs:
```diff
@@ -59,6 +59,23 @@ private void OnDisable()
+		Transform ObjectHolder = CargoShuttle.Instance.SearchForObjectsOnShuttle();
+		foreach (Transform child in ObjectHolder)
+		{
+			if (child.transform.gameObject.layer == 12 || child.transform.gameObject.layer == 8)
```
Can't you just do `child.gameObject.layer`? Because `child` is already `transform`.

ReiderJack replied:
Cache `LayerMask` before `foreach`
`LayerMask layersToCheck = LayerMask.GetMask("Player", "NPC")`

Then do the check `if((1<<child.gameObject.laer & layersToCheck) == 0) continue;`
<https://answers.unity.com/questions/422472/how-can-i-compare-colliders-layer-to-my-layermask.html>

LuisDSG committed:
Update CargoManager.cs

-Edited "CheckLifeforms" to look for lifeforms on the shuttle using LayerMask and edited its if condition to compare against it.
